OICI-213:lot 9 formulaire OP3

Iandraina-etech opened the pull request:
Ajout des champs necessaire pour adaptation de l object claim a fin qu'il puisse prendre en compte les details de l'OP3

date preAuthorizatioon
code specifique a la preAuthorization
-status preAuth
-adaptation deschamps pour qu'il soient non necessaire dans la requette GQL mais qu'il y ai une verification cote back pour que le back (pour les claims classique) , ainsi la creation peut prendre en compte les OP3 et les PEC classiques
-Ajout de permissions pour le nouveau workflow
-MEP du nouveau workflow
-verification des prestation de PreAuth pour les services et items

Iandraina-etech committed:
Update default config for claim panel and diagnosis

Changed ClaimMasterPanel to show patient condition and pre-authorization by default. Set DEFAULT_ADDITIONAL_DIAGNOSIS_NUMBER to 1 in constants.js to adjust default diagnosis entries.

Mirado-etech committed:
Create initial claim duplicates

Mirado-etech committed:
Default Preauthorization Form

Mirado-etech committed:
Update form and create initial searcher

Mirado-etech committed:
Fix filter for pre authorizations

Mirado-etech committed:
Create pages and routing

Mirado-etech committed:
Correct form routing from list

Iandraina-etech committed:
Merge pull request #11 from Iandraina-etech/OICI-213-Mirado

Base for Pre authorization Logic

Iandraina-etech committed:
Add pre-authorization code support for claims

Introduces handling for claim pre-authorization codes across actions, reducer, and UI components. Adds validation, state management, and display logic for pre-authorization codes, including new actions, reducer cases, and updates to forms and searcher components. Also removes some console logs and disables certain fields and checks in preauthorization forms.

Iandraina-etech committed:
Update preauthorization submission logic

Replaces usage of the generic submit action with submitPreAuthorization in PreauthorizationsPage for clarity and specificity. Adds a forced filter for statusPreAuthorization in PreauthorizationSearcher to refine search results.

Iandraina-etech committed:
Add pre-authorization workflow and rejection dialog

Introduces new actions and UI for handling claim pre-authorization workflow, including submitting to medical/normal claim, and rejecting claims with a reason. Adds status tracking for pre-authorizations, updates reducers and constants, and integrates a RejectionDialog component. Also updates translations and searcher to display pre-authorization status.

Iandraina-etech committed:
Merge pull request #12 from Iandraina-etech/OICI-213-Iandraina

Oici 213 iandraina

Mirado-etech committed:
Correct back route

Iandraina-etech committed:
Merge pull request #13 from Iandraina-etech/OICI-213-Iandraina

Correct back route

Iandraina-etech committed:
Update preauthorization submission logic

Refactored submission checks to use 'statusPreAuthorization' instead of 'status'. Deprecated 'submitAll' and related UI actions, and adjusted claim submission logic to improve clarity and maintainability.

Iandraina-etech committed:
Pass preAuth prop to ClaimChildPanel in PreauthorizationForm

Added the preAuth prop to ClaimChildPanel instances in PreauthorizationForm to enable pre-authorization-specific behavior. Updated ClaimChildPanel to accept and forward the preAuth prop to PublishedComponent.

Iandraina-etech committed:
Enable claimed column in preauthorization searcher

Uncomments the 'claimed' field and its formatter in the PreauthorizationSearcher component, making the claimed amount visible in the search results.

Iandraina-etech committed:
Add pre-authorization permission checks to form actions

Introduced new constants for pre-authorization permissions and updated PreauthorizationForm to conditionally render action buttons based on user rights. This ensures only users with the appropriate permissions can reject or validate pre-authorizations.

Iandraina-etech committed:
Merge pull request #16 from Iandraina-etech/develop-comores

Develop comores

Iandraina-etech committed:
Add PreAuthChildPanel and update preauthorization forms

Introduces the new PreAuthChildPanel component to handle pre-authorization items and services, replacing ClaimChildPanel in PreauthorizationForm. Updates PreauthorizationMasterPanel and related logic to enforce read-only state based on statusPreAuthorization, and adjusts button visibility and permissions for pre-authorization validation and rejection.

Iandraina-etech committed:
Merge branch 'OICI-213-Iandraina' of https://github.com/Iandraina-etech/openimis-fe-claim_js into OICI-213-Iandraina

Mirado-etech committed:
Normal claim filter

Claims page doesn't show PreAuthorizations anymore.

Mirado-etech committed:
Default preAuthorization value

Pre authorization default to false when we create a normal claim

Iandraina-etech committed:
Add PreauthorizationFilter and use in PreauthorizationSearcher

Introduced a new PreauthorizationFilter component for advanced filtering of preauthorizations. Updated PreauthorizationSearcher to use PreauthorizationFilter instead of ClaimFilter for its filter pane.

Iandraina-etech committed:
Add PreAuthFilterPicker and integrate pre-authorization status filter

Introduced a new PreAuthFilterPicker component for filtering by pre-authorization status. Updated PreauthorizationFilter to use this picker and added relevant constants and translations. Registered the new picker in the published components.

Iandraina-etech committed:
Revert "Update default config for claim panel and diagnosis"

This reverts commit 34e58cf5608c718d234ed6cf4856cf9eee473bd4.

Iandraina-etech committed:
Merge pull request #17 from Iandraina-etech/OICI-213-Iandraina

Oici 213 iandraina

Iandraina-etech committed:
Add isPreAuthorization field to claim initialization

Iandraina-etech committed:
Merge pull request #18 from Iandraina-etech/OICI-213-Iandraina

Add isPreAuthorization field to claim initialization

Mirado-etech committed:
Merge pull request #19 from Iandraina-etech/OICI-213-Mirado

Normal Claim filter

Iandraina-etech committed:
Merge pull request #20 from Iandraina-etech/OICI-213

OICI 213: lot 9

Mirado-etech committed:
Read only preAuthorizations datas

The datas from Pre Authorizations are read only when it becomes a claim.

Mirado-etech committed:
PreAuthorization call date field logic

vaneck237 commented on src/actions.js:
```diff
@@ -537,7 +564,8 @@ export function formatClaimGQL(modulesManager, claim, shouldAutogenerate) {
   return `
     ${claim.uuid !== undefined && claim.uuid !== null ? `uuid: "${claim.uuid}"` : ""}
-    code: "${isAutogenerateEnabled ? claimCodePlaceholder : claim.code}"
+    ${isAutogenerateEnabled ? `code : "${claimCodePlaceholder}" ` : claim.code?`code : "${claimCodePlaceholder}" `:""}
```
c'est inutile de rajouter un controle sur le claim.code étant donné que c'est un champs obligatoire dans le formulaire

Iandraina-etech replied:
Vu que l'on cree un nouveau formulaire OP3 qui utilise la meme entite maintenant si je cree une OP3 ca sera un code de pre_authorisation et non pas le code de claim classique voila pourquoi j'ai rajoute cette condition ,mais si on insere un claim classique il y  a tout de meme la verification dans la fonction CanSave que le code existe et n'est pas vide

vaneck237 commented on src/components/ClaimForm.js:
```diff
@@ -142,7 +142,9 @@ class ClaimForm extends Component {
     claim.code = "";
+    claim.isPreAuthorization=false;
     claim.preAuthorization = false;
+    claim.isPreAuthorization=false;
```
Pourquoi rajouter ceci une 2e fois? c'est déjà présent a la ligne 145

Iandraina-etech replied:
resolu

vaneck237 commented on src/components/PreauthorizationMasterPanel.js:
```diff
@@ -0,0 +1,605 @@
+        {/* <ControlledField
+          module="claim"
+          id="Claim.claimedDate"
+          field={
```
il faut supprimer si les champs ne sont pas nécéssaire et non commenter

Iandraina-etech replied:
resolu

vaneck237 commented on src/components/RejectionDialog.js:
```diff
@@ -0,0 +1,140 @@
+  };
+
+  componentDidMount() {
+    console.log("RejectionDialog mounted, claim:", this.props.claim);
```
faut retirer les console log

Iandraina-etech replied:
resolu

vaneck237 commented on src/index.js:
```diff
@@ -48,7 +52,9 @@ const ROUTE_CLAIM_FEEDBACK = "claim/feedback";
-const ROUTE_PRESCRIBER_EDIT = "claim/prescriberEdit"
+const ROUTE_PRESCRIBER_EDIT = "claim/prescriberEdit";
+const ROUTE_PREAUTHORIZATIONS = "claim/preAuthorizations";
+const ROUTE_PREAUTHORIZATION_EDIT = "claim/preAuthorizationEdit";
```
pourquoi pas "claim/preAuthorization/edit" plutot?

Iandraina-etech replied:
resolu

Iandraina-etech committed:
Add emergency preauthorization date field

Introduces the datePreAuthorizationEmergency field to claim and preauthorization forms. Updates GraphQL formatting and projections, removes the field from ClaimMasterPanel, and adds a datetime-local input in PreauthorizationMasterPanel for emergency visit types.

Iandraina-etech committed:
Set field to always disabled in ClaimMasterPanel

Replaces dynamic disabling logic with a static 'disabled={true}' prop for the field in ClaimMasterPanel. Ensures the field is always disabled regardless of the previous 'trueReadonly' value.

Iandraina-etech committed:
Merge pull request #21 from Iandraina-etech/OICI-213

Oici 213

Iandraina-etech committed:
Remove redundant pre-authorization check in form

Simplified the emergency pre-authorization date check by removing the unnecessary isPreAuthorization condition in PreauthorizationForm. This streamlines the logic for emergency visit type validation.

Iandraina-etech committed:
Merge pull request #22 from Iandraina-etech/OICI-213

Remove redundant pre-authorization check in form

Iandraina-etech committed:
Remove unused code and update route path

Cleaned up redundant assignments in ClaimForm, removed commented and debug code from PreauthorizationMasterPanel and RejectionDialog, and updated the preauthorization edit route in index.js for consistency.

Iandraina-etech committed:
Merge pull request #23 from Iandraina-etech/OICI-213

Remove unused code and update route path
